includes/wp-cli/class-reset.php

```diff
@@ -0,0 +1,373 @@
+			'relationships'  => 0,
+			'posts'          => 0,
+			'media'          => 0,
+			'models'         => 0,
+		];
+	}
+
+	/**
+	 * Resets ACM by deleting models, taxonomies, taxonomy terms, posts, relationship data and media items relating to ACM models.
+	 *
+	 * ## OPTIONS
+	 *
+	 * [--yes]
+	 * : Skip prompt to confirm deletion.
+	 *
+	 * [--all]
+	 * : Delete all published posts, pages, custom posts and media, not just posts and media associated with ACM models.
+	 *
+	 * ## EXAMPLES
+	 *
+	 *     wp acm reset
+	 *     wp acm reset --yes
+	 *     wp acm reset --all
+	 *     wp acm reset --yes --all
+	 *
+	 * @param array $args Options passed to the command, keyed by integer.
+	 * @param array $assoc_args Options keyed by string.
+	 */
+	public function __invoke( $args, $assoc_args ) {
+		\WP_CLI::confirm( 'Delete ACM data, including posts, media, taxonomies and models?', $assoc_args );
+
+		$delete_all = $assoc_args['all'] ?? false;
+
+		\WP_CLI::log( 'Deleting taxonomy terms.' );
+		$this->delete_taxonomy_terms();
+
+		\WP_CLI::log( 'Deleting ACM taxonomies.' );
+		$this->delete_taxonomies();
+
+		\WP_CLI::log( 'Deleting relationship data.' );
+		$this->delete_relationships();
+
+		\WP_CLI::log( 'Deleting posts.' );
+		$this->delete_posts( $delete_all );
+
+		\WP_CLI::log( 'Deleting media.' );
+		$this->delete_media( $delete_all );
+
+		\WP_CLI::log( 'Deleting models.' );
+		$this->delete_models();
+
+		$this->log_stats();
+		\WP_CLI::success( 'ACM reset complete.' );
+	}
+
+	/**
+	 * Deletes taxonomy terms
+	 */
+	public function delete_taxonomy_terms() {
+		foreach ( $this->taxonomies as $taxonomy ) {
+			$terms = get_terms(
+				[
+					'taxonomy'   => $taxonomy['slug'],
+					'hide_empty' => false,
+				]
+			);
+			foreach ( $terms as $term ) {
+				if ( (bool) wp_delete_term( $term->term_id, $taxonomy['slug'] ) ) {
+					$this->stats['taxonomy_terms']++;
+				}
+			}
+		}
+	}
+
+	/**
+	 * Deletes ACM taxonomies.
+	 */
+	public function delete_taxonomies() {
+		if ( delete_option( self::ACM_TAXONOMY_OPTION ) ) {
+			$this->stats['taxonomies'] = count( (array) $this->taxonomies );
+			$this->taxonomies          = [];
+		}
+	}
+
+	/**
+	 * Deletes all ACM relationship data.
+	 */
+	public function delete_relationships() {
+		global $wpdb;
+
+		$table        = ContentConnect::instance()->get_table( 'p2p' );
+		$post_to_post = esc_sql( $table->get_table_name() );
+
+		$deleted_rows = $wpdb->query("DELETE FROM `{$post_to_post}`;"); // phpcs:ignore -- table name escaped earlier.
+
+		if ( is_int( $deleted_rows ) ) {
+			$this->stats['relationships'] = $deleted_rows;
+		}
+	}
+
+	/**
+	 * Deletes published posts.
+	 *
+	 * @param bool $delete_all Pass true to also delete post types unrelated to
+	 *                         ACM, such as core posts and pages.
+	 */
+	public function delete_posts( $delete_all = false ) {
+		if ( $delete_all ) {
+			$posts = new \WP_Query(
+				[
+					'post_type'      => 'any',
+					'posts_per_page' => -1,
```

Comment:
Noticed draft posts weren't deleted. This would handle that, unless we want to leave them around.

```suggestion
					'posts_per_page' => -1,
					'post_status'    => 'any',
```

Reply:
I think this might end up also deleting attachments as a side-effect, which we probably want to avoid for a delete posts function, but let me double check.

Reply:
Sounds right, and not deleting drafts should be low impact. Let's leave it as is unless you see a clean way that doesn't complicate things.